File: pdlearning/src/web-apps/src/libraries/domain-api/src/calendar/dtos/get-team-member-events-request.ts
import { IGetParams } from '@opal20/infrastructure';

export class GetTeamMemberEventsRequest implements IGetParams {
  [param: string]: string | number | boolean | IGetParams | string[] | boolean[] | number[];
  public rangeStart: string;
  public rangeEnd: string;
  public memberId: string;
}

File: pdlearning/src/microservices/calendar-microservice/Api/Application/RequestDtos/GetTeamMemberEventsRequest.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;

namespace Microservice.Calendar.Application.RequestDtos
{
    public class GetTeamMemberEventsRequest
    {
        [BindProperty(Name = "memberId")]
        public Guid LearnerId { get; set; }

        [Required]
        public DateTime? RangeStart { get; set; }

        [Required]
        public DateTime? RangeEnd { get; set; }
    }
}
